EKrut-Server/src/ekrut/server/db/InventoryItemDAO.java

```diff
@@ -17,21 +18,71 @@ public class InventoryItemDAO {
 
 	private DBController con;
 	private ItemDAO itemDAO;
+	private iItemQuantityFetcher itemQuantityFetcher;
 
 	public InventoryItemDAO(DBController con) {
 		this.con = con;
-		itemDAO = new ItemDAO(con); 
+		this.itemDAO = new ItemDAO(con);
+		this.itemQuantityFetcher = new ItemQuantityFetcher(con);
 	}
 
 	/**
-	 * Fetch a single InventoryItem object from DB, identified by itemId and ekrutLocation.
+	 * Constructs an InventoryItemDAO object with injectable ItemQuantityFetcher
+	 * dependency.
+	 *
+	 * @param con                 a DBController object for interacting with the database
+	 * @param itemQuantityFetcher an iItemQuantityFetcher object for fetching items quantity via sensor
+	 */
+	public InventoryItemDAO(DBController con, iItemQuantityFetcher itemQuantityFetcher) {
+		this.con = con;
+		this.itemDAO = new ItemDAO(con);
+		this.itemQuantityFetcher = itemQuantityFetcher;
+	}
+
+	/**
+	 * Creates a new InventoryItem in the inventory system.
+	 *
+	 * @param itemId        the ID of the new InventoryItem
+	 * @param quantity      the initial quantity of the new InventoryItem
+	 * @param ekrutLocation the ekrut location of the new InventoryItem
+	 * @param itemThreshold the threshold of the new InventoryItem
+	 * @return a boolean indicating whether the creation was successful
+	 * @throws RuntimeException if there is a problem closing the PreparedStatement object
+	 */
+	public boolean CreateInventoryItem(int itemId, int quantity, String ekrutLocation, int itemThreshold) {
+		// Check if InventoryItem Already exists.
+		if (fetchInventoryItem(itemId, ekrutLocation) != null)
```

Comment:
There's no need to do this extra database query. For an inventory item, itemId + ekrutLocation should be a primary key and therefore attempting to insert an existing item will simply fail.